### PR TITLE
Model upload

### PR DESCRIPTION
**Pull request by codegurudummy**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: model-uploader/src/main/java/ai/djl/uploader/MetadataBuilder.java [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+            item.setSha1Hash(getSha1Sum(copiedFile));
+            item.setSize(copiedFile.length());
+            item.setUri(artifactName + '/' + uri);
+            files.put(names[0], item);
+        }
+        return files;
+    }
+
+    private void gzipFile(File file, Path destination) throws IOException {
+        InputStream is = Files.newInputStream(file.toPath());
```
Comment (codegurudummy) [inline model-uploader/src/main/java/ai/djl/uploader/MetadataBuilder.java:486]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *is*. The resource is referenced in statements at the following line: 492. The resource closure statement is at line: 498. There are other execution paths that do not contain closure statements, for example, when InputStream.read throws an exception. Either a) close *is* in a try-finally block or b) close the resource by declaring *is* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).

File: model-uploader/src/main/java/ai/djl/uploader/MetadataBuilder.java [hunk 2]
```diff
@@ -0,0 +1,583 @@
...
+
+        OutputStream os = Files.newOutputStream(destination);
```
Comment (codegurudummy) [inline model-uploader/src/main/java/ai/djl/uploader/MetadataBuilder.java:488]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *os*. The resource is referenced in statements at the following lines: 489, 493. The resource closure statement is at lines: 496, 497. There are other execution paths that do not contain closure statements, for example, when InputStream.read throws an exception. Either a) close *os* in a try-finally block or b) close the resource by declaring *os* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).

File: model-uploader/src/main/java/ai/djl/uploader/MetadataBuilder.java [hunk 3]
```diff
@@ -0,0 +1,583 @@
...
+        GZIPOutputStream gzipOS = new GZIPOutputStream(os);
```
Comment (codegurudummy) [inline model-uploader/src/main/java/ai/djl/uploader/MetadataBuilder.java:489]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *gzipOS*. The resource is referenced in statements at the following lines: 489, 493. The resource closure statement is at lines: 496, 497. There are other execution paths that do not contain closure statements, for example, when InputStream.read throws an exception. Either a) close *gzipOS* in a try-finally block or b) close the resource by declaring *gzipOS* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).
